### DISCUSSION TITLE
I'm too stupid. How to use EasyEffects only for Spotify with a external audio adapter (DAC)?

### DISCUSSION THREAD

**Discussion by fiechr** [wwmm/easyeffects#2377] [Q&A] [answered]
I'm running Manjaro Linux with KDE/Plasma.

Pains me to say it, but even as an IT guy I really have difficulties understanding how EasyEffects actually works inside my existing setup.

I'm using the onboard Realtek sound with some analog speakers and I have a additional headphone DAC (Topping DX1) connected via USB and use this primarily for listening to music with Spotify. Until now, I've set the Spotify application to the DAC using the KDE audio settings, which is remembered and works perfectly fine each time I start it back up again. Everything else uses the regular speakers.

Now I wanted to try out maybe some EQ for the headphones only (at least at first) and installed EasyEffects.

Under Device Management in Pipewire, there is "Use Default Input/Output". But what do I select here? Does this change the Pipewire settings, the EasyEffects settings or something else? If I disable the "Use default output device" options all audio applications, like video players are affected. It's surprising to me, that it influences all sound output of the whole system. Therefore I'm still unsure what to make of this setting and if I need to change it at all. EasyEffects sink is *not* set as default audio device in the KDE settings. The analog audio out for the speakers is set as default device.

How do I set this up with EasyEffects to work only "between" Spotify and my DAC for now and leave my other stuff alone? Am I still allowed to set the output device for Spotify using the KDE settings? Is it possible to limit any influence to a certain application and/or a certain output device, without changing the system settings for the rest of the apps and output devices?

**Comment by wwmm** [comment, marked as answer]
> Does this change the Pipewire settings

No. They only allow the user to change which input/output devices EasyEffects is going to use.

>If I disable the "Use default output device" options all audio applications, like video players are affected. It's surprising to me, that it influences all sound output of the whole system.

EasyEffects doesn't even try to set the system default devices. There is no code for this in it. What it tries when `Process All Output Streams` is enabled in its preferences window is moving the players stream to its virtual devices.

>EasyEffects sink is not set as default audio device in the KDE settings.

Keep your hardware as default. More details about why at #1836.

>How do I set this up with EasyEffects to work only "between" Spotify and my DAC for now and leave my other stuff alone?

First disable `Process All Output Streams` to avoid EasyEffects moving the other streams to its virtual devices. Second disable `Use Default Output` in its PipeWire tab and select your DAC as output. And finally once Spotify is running check its `Enable` checkbox in EasyEffects `Players` tab.

If everything goes well PipeWire/WirePlumber will remember the association done manually between Spotify and our virtual sink. But it may just as well automatically move to it apps that were using our virtual sink in the past. So after setting all of this you may have to manually disable effects for some of these streams at least once until wireplumber forgets the association it did while `Process All Output Streams` was enabled.

>Is it possible to limit any influence to a certain application and/or a certain output device

There is a blocklist. But it will be tedious for the use case you want. It will be easier to disable `Process All Output Streams` and enable manually the apps for which you want effects.

**Comment by fiechr** [comment]
Thank you for your detailed answer! I did like you recommended and everything seems to work fine now!
